core-api/src/main/java/com/sequenceiq/cloudbreak/validation/KerberosRequestValidator.java

```diff
@@ -0,0 +1,27 @@
+package com.sequenceiq.cloudbreak.validation;
+
+import java.util.List;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.sequenceiq.cloudbreak.api.model.kerberos.KerberosRequest;
+
+public class KerberosRequestValidator implements ConstraintValidator<ValidKerberosRequest, KerberosRequest> {
+
+    @Override
+    public boolean isValid(KerberosRequest req, ConstraintValidatorContext constraintValidatorContext) {
+        return isValid(req);
+    }
+
+    public static boolean isValid(KerberosRequest request) {
+        return List.of(request.getActiveDirectory() != null, request.getFreeIpa() != null,
+                    request.getMit() != null, request.getAmbariKerberosDescriptor() != null)
+                .stream()
+                .filter(fieldIsNotNull -> fieldIsNotNull)
```

Comment:
i cannot use `allMatch` here because the validation is about to check whether only one of the four object is not null, with `allMatch` i can only check whether all of them are not null